apps/passport-server/src/services/telegramService.ts

```diff
@@ -990,20 +994,65 @@ export class TelegramService {
     });
   }
 
+  /**
+   * This is called from the /telegram/verify route.
+   */
+  public async handleVerification(
+    serializedPCD: string,
+    telegramUserId: number,
+    telegramChatId: string,
+    telegramUsername?: string
+  ): Promise<void> {
+    return traced("telegram", "handleVerification", async (span) => {
+      const parsed = JSON.parse(serializedPCD) as SerializedPCD;
+      if (parsed.type == ZKEdDSAEventTicketPCDTypeName) {
+        await this.handleTicketVerification(
+          serializedPCD,
+          telegramUserId,
+          telegramChatId,
+          telegramUsername
+        );
+      } else if (parsed.type == ZKEdDSAFrogPCDTypeName) {
+        await handleFrogVerification(
+          serializedPCD,
+          telegramUserId,
+          telegramChatId,
+          telegramUsername
+        );
+      }
+
+      span?.setAttribute("chatId", telegramChatId);
+      const chat = await getGroupChat(this.authBot.api, telegramChatId);
+      span?.setAttribute("chatTitle", chat.title);
+
+      // We've verified that the chat exists, now add the user to our list.
+      // This will be important later when the user requests to join.
+      // TODO (Veronica): semaphore_id???
+      await insertTelegramVerification(
```

Comment:
We have to insert this to the table "telegram_bot_conversations" here. 
Otherwise, the "chat_join_request" on authBot won't work, because it checks this table to see if the user is verified.
However, we cannot insert it to this table here because of the semaphore_id. We are hiding the owner semaphore id for ZKEdDSAFrogPCD.

And semaphore_id has all these constraints
-- Step 1: Add semaphore_id column
ALTER TABLE telegram_bot_conversations
ADD column semaphore_id VARCHAR NOT NULL;

-- Step 2: Add foreign key constraint on users commitment
ALTER TABLE telegram_bot_conversations
ADD CONSTRAINT fk_semaphore_commitment
FOREIGN KEY (semaphore_id) REFERENCES users(commitment) ON UPDATE CASCADE;

-- Step 3: Add a unique index on (semaphore_id, telegram_chat_id)
-- This prevents a user with two Telegram accounts from joining the chat multiple times
CREATE UNIQUE INDEX idx_unique_semaphore_telegram
ON telegram_bot_conversations(semaphore_id, telegram_chat_id);

What should we do here?
Should we add a new table? Or drop some of the constraints here?

Reply:
I'll change the ZKEdDSAFrogPCD to reveal ownerSemaphoreId.